pkg/render/monitor/monitor.go

```diff
@@ -45,35 +48,74 @@ const (
 	CalicoNodePrometheus        = "calico-node-prometheus"
 	ElasticsearchMetrics        = "elasticsearch-metrics"
 	FluentdMetrics              = "fluentd-metrics"
+	TigeraPrometheusObjectName  = "tigera-prometheus"
+	TigeraPrometheusSAName      = "prometheus"
 	TigeraPrometheusDPRate      = "tigera-prometheus-dp-rate"
 	TigeraPrometheusRole        = "tigera-prometheus-role"
 	TigeraPrometheusRoleBinding = "tigera-prometheus-role-binding"
 
-	PrometheusHTTPAPIServiceName = "prometheus-http-api"
-	PrometheusDefaultPort        = 9090
+	PrometheusHTTPAPIServiceName    = "prometheus-http-api"
+	PrometheusDefaultPort           = 9090
+	PrometheusProxyPort             = 9095
+	PrometheusTLSSecretName         = "calico-node-prometheus-tls"
+	calicoNodePrometheusServiceName = "calico-node-prometheus"
+
+	tigeraPrometheusServiceHealthEndpoint = "/health"
 
 	AlertmanagerConfigSecret = "alertmanager-calico-node-alertmanager"
 
 	prometheusServiceAccountName = "prometheus"
 )
 
-func Monitor(cfg *Config) render.Component {
-	return &monitorComponent{
-		cfg: cfg,
+func Monitor(cfg *Config) (render.Component, error) {
+	var err error
+	var tlsSecrets []*corev1.Secret
+	var tlsHash string
+	if cfg.Installation.CertificateManagement == nil {
+		if cfg.TLSSecret == nil {
+			svcDNSNames := dns.GetServiceDNSNames(PrometheusHTTPAPIServiceName, common.TigeraPrometheusNamespace, cfg.ClusterDomain)
+			cfg.TLSSecret, err = secret.CreateTLSSecret(nil,
```

Comment:
Is the customer able to provide the TLS configuration?

Reply:
I'm following the established pattern, which enables a user to provide the certificates. I have not yet added the doc for it. Let me add that to a ticket.